src/tools/rust-analyzer/crates/hir-ty/src/mir/eval.rs

```diff
@@ -1638,6 +1638,7 @@ impl Evaluator<'_> {
             return Ok(0);
         };
         match &layout.variants {
+            Variants::Empty => unreachable!(),
```

Comment:
I don't know *why* this is unreachable, but the old logic would already have panicked here in at least some of same cases since the `variants[index.0]` below would fail for no-variant enums.

Reply:
It seems to be used only in mir evaluator, which I guess would not go into unreachable code.